blocktx/PeerHandler.go

```diff
@@ -290,20 +291,29 @@ func (bs *PeerHandler) HandleBlock(wireMsg wire.Message, peer p2p.PeerI) error {
 		return fmt.Errorf("unable to insert block %s: %v", blockHash.String(), err)
 	}
 
-	if err = bs.markTransactionsAsMined(blockId, msg.TransactionHashes); err != nil {
-		return fmt.Errorf("unable to mark block as mined %s: %v", blockHash.String(), err)
+	transactionHashes := make([]string, len(msg.TransactionHashes))
+	for i, hash := range msg.TransactionHashes {
+		transactionHashes[i] = hash.String()
 	}
 
-	transactionHashes := make([][]byte, len(msg.TransactionHashes))
-	for i, hash := range msg.TransactionHashes {
-		transactionHashes[i] = hash[:]
+	calculatedMerkleRoot, err := bc.BuildMerkleTreeStore(transactionHashes)
```

Comment:
It would be better to call this `calculatedMerkleTree` instead of `calculatedMerkleRoot` as the root is only the last element of that slice